mava/components/training/model_updating.py

```diff
@@ -295,16 +295,11 @@ def model_update_epoch(
 
             base_key, shuffle_key = jax.random.split(key)
 
-            # TODO (dries): This assert is ugly. Is there a better way to do this check?
-            # Maybe using a tree map of some sort?
-            # shapes = jax.tree_util.tree_map(
-            #         lambda x: x.shape[0]==trainer.store.full_batch_size, batch
-            #     )
-            # assert ...
-            assert (
-                list(batch.observations.values())[0].observation.shape[0]
-                == trainer.store.full_batch_size
-            )
+            # Check that our obs have correct batch shape
+            sample_observation = jax.tree_util.tree_leaves(
+                list(batch.observations.values())[0].observation
+            )[0]
+            assert sample_observation.shape[0] == trainer.store.full_batch_size
```

Comment:
Should we maybe loop through and check every observation?

Reply:
I think it is to do only the first for now since I believe these obs within the same batch, have the same shape.

Reply:
Seems like an expensive check to do every update is there a reason it was commented out originally?

Reply:
I am happy to remove this check completely. 

@DriesSmit you okay with this?

Reply:
I far as I understand `jit`, think this is fine. This function is jitted and therefore the assert is only computed once at compile time. See [here](https://github.com/instadeepai/Mava/pull/797/files). I think a loop might be even better as there is no hit to performance.

Reply:
Thanks @DriesSmit ! 

I think the main question here is whether this check gives us any useful information? If not, we should probably remove it.

Reply:
I originally added the check to make sure that the shapes were correct when developing. But now that you mention it, that is what the tests are for. So I am happy to remove it :slightly_smiling_face:

Reply:
That's really cool, if we do have any other asserts in the training/executing loop we should really leave a comment with that link to explain that it's only executed once

Reply:
So I removed this for :)

Reply:
Agreed @sash-a :+1: